src/create-ssr-search-client.ts

```diff
@@ -0,0 +1,109 @@
+import * as algoliasearchProxy from 'algoliasearch';
+import * as encodeProxy from 'querystring-es3/encode';
+import { VERSION as AngularVersion } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { TransferState, StateKey } from '@angular/platform-browser';
+import { VERSION } from './version';
+
+type SSRSearchClientOptions = {
+  appId: string;
+  apiKey: string;
+  httpClient: HttpClient;
+  HttpHeaders: typeof HttpHeaders;
+  transferState: TransferState;
+  makeStateKey<T = void>(key: string): StateKey<T>;
+};
+
+type RequestOptions = {
+  method:
+    | 'GET'
```

Comment:
since this is used for Algolia alone, you can keep only POST, GET. There's more used, but those are for the indexing methods

Reply:
Indeed – here's the reference: https://www.algolia.com/doc/rest-api/search/#search-endpoints.